v2/authorcontext/author_context.go

```diff
@@ -0,0 +1,18 @@
+package authorcontext
```

Comment:
Maybe requester is even a better name if want to change it. Because we use author in users repo for the requester who changes the things. It is the same here as well. It can be impersonator or user itself. 

Since I wanted to be consistent with the users repo I named as "Author"

Reply:
Requester of what? For the service that uses the service this information should only be used if they want to know if the user in the User ID context is who the token says he is.

I dont think what you use internally in Users is relevant here. The author of the request might make sense there, but in general, we need a better, non-confusing name.

Reply:
An alternative idea would be to include this functionality into the user context instead. And having some access function to figure out if the user has been impersonated, and by who. Would remove the need of the extra context

Reply:
Requester is the owner of the request who made a call to an end point. Since this is a middleware and works for every request for the miscroservices which use this middleware. Anyway, for now I guess Author is more clear.

For the User context we already made the changes by wrapping jwt package and it recognizes whether user has been impersonated or not. But still there is a problem for other microservices which uses this middleware or go-enlight-middleware. Since both have a memory cache which caches user's Enlight ID and don't call the SSO api if it's been cached. 

With adding extra parameter to the context will make life easier for all microservices otherwise they will need to parse the token again to understand  whether it's impersonated. Even it's good for Users context to reduce some cycles.